src/components/Header.tsx

```diff
@@ -0,0 +1,127 @@
+import { useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+import { showError } from '../helpers/helpers';
+import { updateTodo, addTodos } from '../api/todos';
+
+interface Props {
+  todos: Todo[],
+  setSearchQuery: (SearchQuery: string) => void;
+  searchQuery: string;
+  setError: React.Dispatch<React.SetStateAction<string>>;
+  setTempTodo: (tempTodo: Todo | null) => void;
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  setLoader: React.Dispatch<React.SetStateAction<number[]>>;
+}
+
+export const Header: React.FC<Props> = ({
+  todos,
+  setSearchQuery,
+  searchQuery,
+  setError,
+  setTempTodo,
+  setTodos,
+  setLoader,
+}) => {
+  const [submitDisable, setSubmitDisable] = useState(false);
+  const activeInputButton = todos.every(todo => todo.completed);
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    (setSearchQuery(event.target.value));
+  };
+
+  const newTodo = {
+    id: 0,
+    title: searchQuery,
+    userId: 10881,
+    completed: false,
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setSubmitDisable(true);
+    if (!searchQuery) {
+      showError('Title can\'t be empty', setError);
+      setSubmitDisable(false);
+    } else {
+      addTodos(10881, searchQuery)
+        .then(todo => {
+          setTodos(prevState => {
+            setTempTodo(null);
+            setSubmitDisable(false);
+            setSearchQuery('');
+
+            return [...prevState, todo];
+          });
+        })
+        .catch(() => showError('Unable to add a todo', setError));
+      setTempTodo(newTodo);
+    }
+  };
+
+  const handleClick = () => {
+    if (activeInputButton) {
+      todos.forEach(todo => {
+        setLoader(prevState => [...prevState, todo.id]);
+        updateTodo(todo.id, { completed: !todo.completed })
+          .then((updatedTodo) => {
+            setTodos(prevState => (
+              prevState.map(prevTodo => {
+                return (prevTodo.id === todo.id) ? updatedTodo : prevTodo;
+              })
+            ));
+
+            setLoader(prevState => prevState.filter(id => id !== todo.id));
+          })
+          .catch(() => showError('Unable to update a todo', setError));
+      });
+    } else {
+      todos.forEach(todo => {
+        if (!todo.completed) {
+          setLoader(prevState => [...prevState, todo.id]);
+          updateTodo(todo.id, { completed: !todo.completed })
+            .then((updatedTodo) => {
+              setTodos(prevState => (
+                prevState.map(prevTodo => {
+                  return (prevTodo.id === todo.id) ? updatedTodo : prevTodo;
+                })
+              ));
+
+              setLoader(prevState => prevState.filter(id => id !== todo.id));
+            })
+            .catch(() => showError('Unable to update a todo', setError));
+        }
+      });
+    }
+  };
```

Comment:
handleClick is not the best way to describe what your button meant for

this code looks really messy, please, think how you can fix it, to get rid of all these inner scopes